src/main/java/com/uber/jenkins/phabricator/PhabricatorNotifier.java

```diff
@@ -188,21 +195,15 @@ private CoverageProvider getUberallsCoverage(AbstractBuild build, BuildListener
         }
 
         Logger logger = new Logger(listener.getLogger());
-
-        Provider<CoverageProvider> provider = new BaseProvider<CoverageProvider>(
+        InstanceProvider<CoverageProvider> provider = new InstanceProvider<CoverageProvider>(
```

Comment:
What about a common method to get instances of various providers easier? This can probably live in InstanceProvider. It can look something like:

public static T getInstance(JenkinsInstance, String pluginName, String pluginClass, Logger logger) {
 // common code
}